sklearn/feature_extraction/hashing.py

```diff
@@ -86,8 +92,9 @@ class FeatureHasher(BaseEstimator, TransformerMixin):
     """
 
     def __init__(self, n_features=(2 ** 20), input_type="dict",
-                 dtype=np.float64, alternate_sign=True, non_negative=False):
-        self._validate_params(n_features, input_type)
```

Comment:
I know it's not the case atm, but validation really needs to be in fit

Reply:
I tried removing this but there are certain test cases that fail on removing this